starfish/display/stack.py

```diff
@@ -0,0 +1,222 @@
+    This function ensures that the output is instead (features, 1, rounds), by inserting singleton
+    dimensions in place of any projected axes.
+
+    Parameters
+    ----------
+    intensity_table : IntensityTable
+        Input intensities
+    dimensions : Set[Indices]
+        Dimensions to project
+
+    Returns
+    -------
+    IntensityTable :
+        full-dimensionality (3-d) IntensityTable
+
+    """
+    str_dimensions = _normalize_indices(dimensions)
+    initial_dimensions = OrderedDict(intensity_table.sizes)
+    projected_intensities = intensity_table.max(str_dimensions)
+    expanded_intensities = projected_intensities.expand_dims(str_dimensions)
+    return expanded_intensities.transpose(*tuple(initial_dimensions.keys()))
+
+
+def _mask_low_intensity_spots(
+    intensity_table: IntensityTable,
+    intensity_threshold: float
+) -> np.ndarray:
+    """
+    returns a flattened boolean mask (c order flattening) that is True where
+    (feature, round, channel) combinations whose intensities are above or equal to intensity
+    threshold and where values are not nan
+    """
+    with warnings.catch_warnings():
+        # we expect to have invalid values (np.nan) in IntensityTable, hide the warnings from users
+        warnings.simplefilter('ignore', RuntimeWarning)
+        return np.ravel(intensity_table.values >= intensity_threshold)
+
+
+def _spots_to_markers(intensity_table: IntensityTable) -> Tuple[np.ndarray, np.ndarray]:
+    """
+    convert each (r, c) combination for each spot into a 5-tuple (x, y, r, c, z) for plotting
+    with napari.
+    """
+    # get sizes which will be used to create the coordinates that correspond to the spots
+    n_rounds = intensity_table.sizes[Indices.ROUND.value]
+    n_channels = intensity_table.sizes[Indices.CH.value]
+    n_features = intensity_table.sizes[Features.AXIS]
+    code_length = n_rounds * n_channels
+
+    # create 5-d coordinates for plotting in (x, y, round. channel, z)
+    n_markers = np.product(intensity_table.shape)
+    coords = np.zeros((n_markers, 5), dtype=np.uint16)
+
+    # create the coordinates.
+    # X, Y, and Z are repeated once per (r, ch) pair (code_length).
+    # the cartesian product of (r, c) are created, and tiled once per feature (n_features)
+    # we ensure that (ch, round) cycle in c-order to match the order of the linearized
+    # array, used below for masking.
+    coords[:, 0] = np.repeat(intensity_table[Features.AXIS][Indices.X.value].values, code_length)
+    coords[:, 1] = np.repeat(intensity_table[Features.AXIS][Indices.Y.value].values, code_length)
+    coords[:, 2] = np.tile(np.tile(range(n_rounds), n_channels), n_features)
+    coords[:, 3] = np.tile(np.repeat(range(n_channels), n_rounds), n_features)
+    coords[:, 4] = np.repeat(intensity_table[Features.AXIS][Indices.Z.value].values, code_length)
+
+    sizes = np.repeat(intensity_table.radius.values, code_length)
+
+    return coords, sizes
+
+
+def stack(
```

Comment:
maybe we should explicitly call this napari_stack, I think we use stack in a lot of other places synonymous with imagestack so this was confusing to me at first

Reply:
Well, we _are_ showing an ImageStack here? I can call this `display_stack` if you'd like. Are you OK with the import pattern of `starfish.display.stack`? Are there alternatives you'd prefer? I'm not wed to this. :-)

Reply:
When I first read this, I thought a super-cool way to do this would be a starfish.display(..) method that takes a variety of objects and just knows how to display them.  But the other display methods seem to require additional parameters, so it's less clear how that strategy would work.

Reply:
Ah yea starfish.display.dispay_stack would be weird. I think I misunderstood the organization, we're you thinking starfish.display.stack displays an imagestack and starfish.display.whatever would display something else? If so I'm on board with that